src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1264,6 +1304,85 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
+        flow1.source.detectConnectedDevices.arp = true
+        flow1.source.vlanId = commonOuterVlanId
+        flow1.source.innerVlanId = innerVlanIdFlow1
+        flow1.encapsulationType = encapsulationType
+        flowHelperV2.addFlow(flow1)
+
+        def flow2 = flowHelperV2.singleSwitchFlow(sw, true, [flow1])
+        flow2.source.detectConnectedDevices.lldp = true
+        flow2.source.detectConnectedDevices.arp = true
+        flow2.source.vlanId = commonOuterVlanId
+        flow2.source.innerVlanId = innerVlanIdFlow2
+        flow2.encapsulationType = encapsulationType
+        flowHelperV2.addFlow(flow2)
+
+        when: "Device connects to src endpoint and send lldp and arp packets for flow1 only"
+        def lldpData = LldpData.buildRandom()
+        def arpData = ArpData.buildRandom()
+        new ConnectedDevice(traffExamProvider.get(), topology.getTraffGen(sw.dpId),
+                [flow1.source.vlanId, flow1.source.innerVlanId]).withCloseable {
+            it.sendLldp(lldpData)
+            it.sendArp(arpData)
+        }
+
+        then: "LLDP and ARP connected devices are recognized for flow1"
+        Wrappers.wait(WAIT_OFFSET) { //need some time for devices to appear
+            verifyAll(northbound.getFlowConnectedDevices(flow1.flowId)) {
+                it.source.lldp.size() == 1
+                it.source.arp.size() == 1
+                it.destination.lldp.empty
+                it.destination.arp.empty
+                verifyEquals(it.source.lldp[0], lldpData)
+                verifyEquals(it.source.arp[0], arpData)
+            }
+        }
+
+        and: "LLDP and ARP connected devices are not recognized for flow2"
+        verifyAll(northbound.getFlowConnectedDevices(flow2.flowId)) {
+            it.source.lldp.empty
+            it.source.arp.empty
+            it.destination.lldp.empty
+            it.destination.arp.empty
+        }
+
+        and: "Devices are registered on the switch"
+        verifyAll(northboundV2.getConnectedDevices(flow1.source.switchId).ports) {
+            it.size() == 1
+            it[0].portNumber == flow1.source.portNumber
+            it[0].lldp.size() == 1
+            it[0].lldp.first().flowId == flow1.flowId
+            it[0].lldp.first().vlan == commonOuterVlanId //due to issue 3475
+            verifyEquals(it[0].lldp.first(), lldpData)
+            it[0].arp.size() == 1
+            it[0].arp.first().flowId == flow1.flowId
+            it[0].arp.first().vlan == commonOuterVlanId //due to issue 3475
+            verifyEquals(it[0].arp.first(), arpData)
+        }
+
+        cleanup: "Restore initial switch properties"
+        flow1 && flowHelperV2.deleteFlow(flow1.flowId)
+        flow2 && flowHelperV2.deleteFlow(flow2.flowId)
+        initialProps && restoreSwitchProperties(sw.dpId, initialProps)
+        sw && database.removeConnectedDevices(sw.dpId)
+
+        where:
+        commonOuterVlanId | innerVlanIdFlow1 | innerVlanIdFlow2 | encapsulationType
```

Comment:
this is fine I guess, but the only thing that actually changes between iterations is `encapsulationType`

Reply:
you are right,
I just didn't want to extract them into separate variables or use magic numbers.